### PR TITLE
Project deaths

### PR DESCRIPTION
**Pull request by Eleonore9**

### CODE REVIEW

File: test/witan/models/dem/ccm/mort/.#mortality_mvp_test.clj [hunk 1]
```diff
@@ -0,0 +1 @@
+eleonore@eleonore-XPS13-9333.7330:1465807268
```
Comment (acron0) [inline test/witan/models/dem/ccm/mort/.#mortality_mvp_test.clj]: Think this file is here by mistake

File: src/witan/models/dem/ccm/mort/mortality_mvp.clj [hunk 2]
```diff
@@ -48,6 +48,18 @@
   [{:keys [historic-asmr]} {:keys [number-of-years-mort jumpoff-year-mort]}]
   {:initial-projected-mortality-rates (cf/jumpoffyr-method-average historic-asmr
                                                                    :death-rate
-                                                                   :proj-death-rate
+                                                                   :death-rate
                                                                    number-of-years-mort
                                                                    jumpoff-year-mort)})
+
+(defworkflowfn project-deaths-from-fixed-rates
```
Comment (acron0) [inline src/witan/models/dem/ccm/mort/mortality_mvp.clj:72]: Please add a docstring. Considering making this mandatory...

Comment (LoraArmstrong) [inline src/witan/models/dem/ccm/mort/mortality_mvp.clj:72]: Sorry about that- should know better.

**Comment by tcoupland** [comment]
:shipit:

File: src/witan/models/dem/ccm/components_functions.clj [hunk 3]
```diff
@@ -19,6 +20,19 @@
         (ds/rename-columns {col-to-avg avg-name})
         (wds/rollup :mean avg-name [:gss-code :sex :age]))))
 
+;; Project component for fixed rates:
+(defn project-component-fixed-rates
+  "Takes in a population at risk and fixed rates for a component.
+  Outputs the population at risk with new columns for the rates
+  and the result of applying the rates to that component."
+  [population-at-risk fixed-rates col-fixed-rates  col-result]
```
Comment (sunnytownsend) [inline src/witan/models/dem/ccm/components_functions.clj]: remove space before col-result

File: src/witan/models/dem/ccm/mort/mortality_mvp.clj [hunk 4]
```diff
@@ -8,6 +8,10 @@
             [incanter.core :as i]))
 
 (defn create-popn-at-risk
+  "Takes datasets with population, deaths, and births. Returns a new
```
Comment (sunnytownsend) [inline src/witan/models/dem/ccm/mort/mortality_mvp.clj]: add to beginning: "Creates population at risk for calculating historic age specific mortality rates."

File: src/witan/models/dem/ccm/mort/mortality_mvp.clj [hunk 5]
```diff
@@ -8,6 +8,10 @@
...
+  dataset with population at risk in the :popn column. The population
+  at risk is the births data joined with a subset of the population
+  dataset that has been aged on."
```
Comment (sunnytownsend) [inline src/witan/models/dem/ccm/mort/mortality_mvp.clj]: The population at risk is the births data (age 0) joined with the aged on population
dataset (age 1 to 90). Pull out popn data for years less than or equal to the latest year of deaths data"

Comment (sunnytownsend) [inline src/witan/models/dem/ccm/mort/mortality_mvp.clj]: move line 21(query dataset) to below line 18 (inc year) - reduces amount of code to do remaining fns on - see if it breaks test!

File: src/witan/models/dem/ccm/mort/mortality_mvp.clj [hunk 6]
```diff
@@ -20,6 +24,9 @@
         (ds/join-rows popn-not-age-0))))
 
 (defn calc-death-rates
+  "Takes datasets with deaths and with population at risk. Calculates
```
Comment (sunnytownsend) [inline src/witan/models/dem/ccm/mort/mortality_mvp.clj]: Add to beginning. Calculates historic age specific mortality rates (ASMR).

File: src/witan/models/dem/ccm/mort/mortality_mvp.clj [hunk 7]
```diff
@@ -40,6 +50,11 @@
        (hash-map :historic-asmr)))
 
 (defworkflowfn project-asmr
+  "Takes a dataset with historic mortality rates, and parameters for
+  the number of years of data to average for the calculation, and the
+  jumpoff year. Returns a dataset that that now includes projected
```
Comment (sunnytownsend) [inline src/witan/models/dem/ccm/mort/mortality_mvp.clj]: double that

File: test/witan/models/dem/ccm/core/projection_loop_test.clj [hunk 8]
```diff
@@ -19,7 +19,7 @@
                     "resources/test_data/handmade_outputs/bristol_migration_module_handmade_output.csv"})))
 
 (def params {:first-proj-year 2014
-             :last-proj-year 2015})
+             :last-proj-year 2016})
```
Comment (sunnytownsend) [inline test/witan/models/dem/ccm/core/projection_loop_test.clj]: remove this as not being used just now

**Comment by sunnytownsend** [comment]
:shipit:

**Pull request merged**